pkg/comp-functions/functions/vshnpostgres/objectbucket.go

```diff
@@ -0,0 +1,36 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+
+	xfnproto "github.com/crossplane/function-sdk-go/proto/v1beta1"
+	appcatv1 "github.com/vshn/appcat/v4/apis/v1"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+)
+
+// EnsureObjectBucketLabels just gets the bucket present from the PnT part and adds it again to the
+// desired state. This ensures that the correct labels are injected.
+func EnsureObjectBucketLabels(ctx context.Context, svc *runtime.ServiceRuntime) *xfnproto.Result {
+
+	comp := &vshnv1.VSHNPostgreSQL{}
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		return runtime.NewFatalResult(fmt.Errorf("Cannot get composite from function io: %w", err))
+	}
+
+	bucket := &appcatv1.XObjectBucket{}
+
+	err = svc.GetDesiredComposedResourceByName(bucket, "pg-bucket")
```

Comment:
We could add a wrapper function here, it's confusing to read this code, you get and the set the same object :)

Reply:
I don't quite follow. How would a wrapper make this more readable?

This function is only necessary as long as we still have the PostgreSQL PnT composition, afterward it will be obsolete.

Reply:
Never mind, it's ok like that. The function description is enough for now.